src/neptune/attributes/series/series.py

```diff
@@ -116,30 +108,26 @@ def log(
         **kwargs,
     ) -> None:
         """log is a deprecated method, this code should be removed in future"""
-        from_stringify_value = False
-        if isinstance(value, StringifyValue):
-            from_stringify_value, value = True, value.value
-
-        if is_collection(value):
-            if step is not None and len(value) > 1:
-                raise ValueError("Collection of values are not supported for explicitly defined 'step'.")
-        else:
-            value = [value]
-
-        if from_stringify_value:
-            value = stringify_unsupported(value)
-
-        value = self._data_to_value(value, **kwargs)
-
         if step is not None:
             verify_type("step", step, (float, int))
         if timestamp is not None:
             verify_type("timestamp", timestamp, (float, int))
 
-        steps = None if step is None else [step]
-        timestamps = None if timestamp is None else [timestamp] * len(value)
+        if is_stringify_value(value):
+            value = self._handle_stringified_value(value)
 
-        ops = self._get_log_operations_from_value(value, steps=steps, timestamps=timestamps)
+        if is_collection(value):
+            if step is not None and len(value) > 1:
+                raise ValueError("Collection of values are not supported for explicitly defined 'step'.")
```

Comment:
Verb to agree with "collection"

```suggestion
                raise ValueError("Collection of values is not supported for explicitly defined 'step'.")
```